lib/iris/fileformats/netcdf.py

```diff
@@ -1938,16 +1939,20 @@ def set_packing_ncattrs(cfvar):
             # Explicitly assign the fill_value, which will be the type default
             # in the case of an unmasked array.
             if packing is None:
-                fill_value = cube.lazy_data().fill_value
+                fill_value = cube.fill_value
                 dtype = cube.lazy_data().dtype.newbyteorder('=')
 
             cf_var = self._dataset.createVariable(
                 cf_name, dtype,
                 dimension_names, fill_value=fill_value,
                 **kwargs)
             set_packing_ncattrs(cf_var)
-            # stream the data
-            biggus.save([cube.lazy_data()], [cf_var], masked=True)
+
+            # Now stream the cube data payload straight to the netCDF
+            # data variable within the netCDF file.
+            data = da.map_blocks(convert_nans_array, cube.lazy_data(),
+                                 nans=cube.fill_value, result_dtype=cube.dtype)
+            da.store([data], [cf_var])
```

Comment:
This is the core change to this PR. Dask streamed saving, using the new `convert_nans_array` utility function to convert the NaN array chunks on the fly as they stream through dask and into disk.